### PR TITLE
fix loadbalancer status comparison

### PR DESCRIPTION
**Pull request by aojea**
The loadbalancer status has added new fields during the latest releases, but the helper function used by the service load balancer controller was not updated with all the new fields, and for the new IPMode field it was not taking into consideration that the field is a pointer.

Instead of checking fields one by one  use the DeepEqual function that provides semantic equality for these types.

/kind bug
/kind cleanup

```release-note
The Service LoadBalancer controller was not correctly considering the service.Status new IPMode field and excluding the Ports when comparing if the status has changed, causing that changes in these fields may not update the service.Status correctly
```

/sig network

### CODE REVIEW

File: staging/src/k8s.io/cloud-provider/service/helpers/helper_test.go [hunk 1]
```diff
@@ -347,3 +348,152 @@ func Test_getPatchBytes(t *testing.T) {
...
+			},
+			r: &v1.LoadBalancerStatus{
+				Ingress: []v1.LoadBalancerIngress{
+					{IP: "192.168.0.1"},
+				},
+			},
+			want: false,
+		},
+		{
+			name: "same ipmode",
```
Comment (aojea) [inline staging/src/k8s.io/cloud-provider/service/helpers/helper_test.go:394]: this failed before, as are different pointer values

**Review by thockin — COMMENTED** [review]
I am not sure why it was done this way.  A little fishy, but I can't see why this would be wrong.

/lgtm
/approve

**Pull request merged**